pkg/graph/resolvers/system_intake_grb_review_form.go

```diff
@@ -100,3 +102,34 @@ func UpdateSystemIntakeGRBReviewFormInputTimeframeAsync(
 		SystemIntake: updatedIntake,
 	}, nil
 }
+
+// PointerToSystemIntakeGRBReviewAsyncStatusType Create a reference to an Async GRB Review status
+func PointerToSystemIntakeGRBReviewAsyncStatusType(
+	status models.SystemIntakeGRBReviewAsyncStatusType,
+) *models.SystemIntakeGRBReviewAsyncStatusType {
+	return &status
+}
+
+// CalcSystemIntakeGRBReviewAsyncStatus calculates the status of the GRB Review Async page
+func CalcSystemIntakeGRBReviewAsyncStatus(
+	intake *models.SystemIntake,
+) (*models.SystemIntakeGRBReviewAsyncStatusType, error) {
```

Comment:
@OddTomBrooks , what do you think about not making this return an error? I think this might be difficult to use in practice on the system intake, especially since the object can be returned as an array. I suspect it might be better to have a default status that returns in those cases instead of error. What do you think?

Otherwise, the logic here looks good.

Reply:
From the standpoint of this function, I agree that trying to "calculate the async status" of something that's not an async review is understandably an error.

I think practically, though, that the FE will be querying this before it knows if the review is standard or async, so it'll probably just be throwing lots of errors and spitting out lots of logs.

This function _already_ returns a pointer to the status, and it's an optional return type in the GQL. Maybe we just return `nil,nil` to communicate more accurately "This intake doesn't _have_ an async review status, because it's not async at all!

That way you don't have to settle for a default cause that doesn't make sense, but can still communicate the same information that an error communicates!

Reply:
That makes sense. Honestly, I still think it's probably better to design atomically around the functional level, but it's a fair compromise.

Reply:
I agree! I think nil, nil would be favorable here in practice. You ok to make the change @OddTomBrooks?

Reply:
Yep. Changes already made. Updating unit tests.